src/service/chat.js

```diff
@@ -232,9 +231,19 @@ const initChatFromPost = async ({source, id}) => {
   }
 };
 
-const getAllowAnonDmStatus = async (postId) => {
+const getAllowAnonDmStatus = async (type, id) => {
+  const generateUrl = () => {
+    switch (type) {
+      case 'post':
+        return `/users/check-allow-dm/?source=post&post_id=${id}`;
+      case 'comment':
+        return `/users/check-allow-dm/?source=comment&comment_id=${id}`;
+      default:
+        return '';
+    }
+  };
   try {
-    const response = await api.get(`/users/check-allow-dm/?source=post&post_id=${postId}`);
+    const response = await api.get(generateUrl());
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The function `getAllowAnonDmStatus` has been refactored to handle different types of sources (post, comment). The URL is now generated dynamically based on the type. This change improves the flexibility of the function but ensure that the correct parameters are passed when calling this function. Also, consider handling the case where an invalid type is passed. Currently, it returns an empty string which might result in an invalid API call.

```diff
- const getAllowAnonDmStatus = async (postId) => {
+ const getAllowAnonDmStatus = async (type, id) => {
+   const generateUrl = () => {
+     switch (type) {
+       case 'post':
+         return `/users/check-allow-dm/?source=post&post_id=${id}`;
+       case 'comment':
+         return `/users/check-allow-dm/?source=comment&comment_id=${id}`;
+       default:
+         return '';
+     }
+   };
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->